src/components/resource-loader/resource-loader.tsx

```diff
@@ -58,3 +58,25 @@ export function loadDeck(deckId: string | undefined, allowUndefinedDeckId = fals
     return deck;
   };
 }
+
+export function loadPubicDeck(deckId?: string) {
+  // hooks outside of functional components must be consumed immediately by the component
+  const { getPublicDeckById } = useDecksClient();
+  const { getPublicCardsById } = useCardsClient(); // TODO: A big error happens when this useCardsClient hook is removed
```

Comment:
the big error should only occur if adding/removing hooks during hot reload right?
i think during development `resource-loader` is difficult since its closure based, so modifying hook order makes it stale and throw that error

but once you're not touching it anymore, the order of hooks is deterministic and the error shouldn't occur

Reply:
also nit: spelling on function name 😆

Reply:
The error was happening when I was navigating from a public deck to the newly copied deck. I believe React doesn't see two separate `ResourceLoader` components, but one `ResourceLoader` with changing hooks.

It is fine now that the hooks are the same between the two helper functions, but if we ever need to add additional hooks in the future, we might run into issues.

Reply:
> also nit: spelling on function name 😆

:joy: it's fixed now